editor/src/components/assets.ts

```diff
@@ -76,87 +77,78 @@ export function projectContentFile(
 
 export type ProjectContentsTree = ProjectContentDirectory | ProjectContentFile
 
-export function mergeTrees(
-  first: ProjectContentTreeRoot,
-  second: ProjectContentTreeRoot,
-): ProjectContentTreeRoot {
-  let merged: ProjectContentTreeRoot = {}
-  Utils.fastForEach(Object.keys(first), (firstKey) => {
-    const fromFirst = first[firstKey]
-    if (firstKey in second) {
-      const fromSecond = second[firstKey]
-      if (fromFirst.type === 'DIRECTORY') {
-        if (fromSecond.type === 'DIRECTORY') {
-          merged[firstKey] = projectContentDirectory(
-            fromSecond.fullPath,
-            fromSecond.directory,
-            mergeTrees(fromFirst.children, fromSecond.children),
-          )
-        } else {
-          throw new Error(
-            `Unable to merge ${JSON.stringify(fromSecond)} into ${JSON.stringify(fromFirst)}`,
-          )
-        }
-      } else {
-        if (fromSecond.type === 'FILE') {
-          merged[firstKey] = fromSecond
-        } else {
-          throw new Error(
-            `Unable to merge ${JSON.stringify(fromSecond)} into ${JSON.stringify(fromFirst)}`,
-          )
-        }
-      }
-    } else {
-      merged[firstKey] = fromFirst
-    }
-  })
-  Utils.fastForEach(Object.keys(second), (secondKey) => {
-    if (!(secondKey in merged)) {
-      merged[secondKey] = second[secondKey]
-    }
-  })
-  return merged
-}
-
 export function getProjectContentKeyPathElements(projectContentKey: string): Array<string> {
   return dropLeadingSlash(projectContentKey)
     .split('/')
     .filter((s) => s.length > 0)
 }
 
-export function pathElementsToProjectContentKey(elements: Array<string>): string {
-  return '/' + elements.join('/')
+export function pathElementsToProjectContentKey(
+  pathElements: Array<string> | ReadonlyArray<string>,
```

Comment:
```suggestion
  pathElements: ReadonlyArray<string>,
```
No need for the union type here. You can pass in an argument of type `Array<T>` to a parameter of type `ReadonlyArray<T>`, but not vice versa.

Reply:
So I did have `ReadonlyArray<string>` in there originally, but TypeScript had a wobbly about my passing a regular `Array<string>` into it at one point.

Reply:
I just shrugged my shoulders, added that in and got on with my day.

Reply:
Weird. I just tested re-writing something similar and it worked fine. It might have been that during one iteration you were doing something that relied on the type of the param matching the type passed in and that's no longer a requirement :shrug:

Reply:
Yeah, it surprised me, but yeah shouldn't need to be a thing.